lib/livebook/file_system/s3.ex

```diff
@@ -24,43 +24,11 @@ defmodule Livebook.FileSystem.S3 do
   end
 
   @doc """
-  Returns a new file system struct.
-
-  ## Options
-
-    * `:region` - the bucket region. By default the URL is assumed
-      to have the format `*.[region].[rootdomain].com` and the region
-      is inferred from that URL
-
-    * `:external_id` - the external id from Teams.
-
-    * `:hub_id` - the Hub id.
-
-    * `:id` - the file system id.
-
+  Returns the region from given uri.
   """
-  @spec new(String.t(), String.t(), String.t(), keyword()) :: t()
-  def new(bucket_url, access_key_id, secret_access_key, opts \\ []) do
-    opts = Keyword.validate!(opts, [:region, :external_id, :hub_id, :id])
-
-    bucket_url = String.trim_trailing(bucket_url, "/")
-    region = opts[:region] || region_from_uri(bucket_url)
-
-    hub_id = Keyword.get(opts, :hub_id, Livebook.Hubs.Personal.id())
-    id = opts[:id] || id(hub_id, bucket_url)
-
-    %__MODULE__{
-      id: id,
-      bucket_url: bucket_url,
-      external_id: opts[:external_id],
-      region: region,
-      access_key_id: access_key_id,
-      secret_access_key: secret_access_key,
-      hub_id: hub_id
-    }
-  end
-
-  defp region_from_uri(uri) do
+  @spec region_from_uri(String.t()) :: String.t()
+  # TODO: make it private again on Livebook v0.12
```

Comment:
Only until we have everything migrated, because the `changeset` will do this job without making it public. That's the only reason to add it to `load/2`. WDYT?

Reply:
Ah yeah, actually since we migrate, we should never expect it to be nil on `load/2`?

Reply:
Yeah, that's the expectation

Reply:
So it should be safe to remove the conditionals from `load/2` :)